web/transfer_controller.go

```diff
@@ -24,15 +27,23 @@ func (tc *TransfersController) Create(c *gin.Context) {
 
 	if err := c.ShouldBindJSON(&tr); err != nil {
 		publicError(c, 400, err)
-	} else if tr.FromAddress != utils.ZeroAddress {
-		if tx, err := store.TxManager.CreateTxWithEth(tr.DestinationAddress, tr.Amount, tr.FromAddress); err != nil {
-			publicError(c, 400, fmt.Errorf("Transaction failed: %v", err))
-		} else {
-			c.JSON(200, tx)
-		}
-	} else if tx, err := store.TxManager.CreateTxWithEth(tr.DestinationAddress, tr.Amount); err != nil {
+	} else if from, err := nextAccount(tr.FromAddress, store); err != nil {
```

Comment:
Let's change the name of the `nextAccount` function to something like `retrieveFromAddress`, then we're good to go! Thanks @whatsadebugger 👍